src/city/view.c

```diff
@@ -581,17 +592,17 @@ void city_view_foreach_tile_in_range(int grid_offset, int size, int radius, map_
         do_valid_callback(x, y, grid_offset, callback);
     }
     for (int ring = 0; ring < radius; ++ring) {
-        int offset_north = grid_offset + map_grid_delta(-ring - 2, -ring - 2);
-        int offset_south = grid_offset + map_grid_delta(ring + size, ring + size);
-        do_valid_callback(x, y + y_offset, offset_south, callback);
-        do_valid_callback(x, y - y_offset, offset_north, callback);
-        do_valid_callback(x - x_offset - x_delta, y, grid_offset + map_grid_delta(-ring - 2, ring + size), callback);
-        do_valid_callback(x + x_offset + x_delta, y, grid_offset + map_grid_delta(ring + size, -ring - 2), callback);
+        int offset_north = -ring - 2;
+        int offset_south = ring + size;
+        do_valid_callback(x, y + y_offset, move_grid_offset_by_delta(grid_offset, ring + size, ring + size), callback);
+        do_valid_callback(x, y - y_offset, move_grid_offset_by_delta(grid_offset, -ring - 2, -ring - 2), callback);
+        do_valid_callback(x - x_offset - x_delta, y, move_grid_offset_by_delta(grid_offset, -ring - 2, ring + size), callback);
+        do_valid_callback(x + x_offset + x_delta, y, move_grid_offset_by_delta(grid_offset, ring + size, -ring - 2), callback);
         for (int tile = 1; tile < ring * 2 + size + 2; ++tile) {
-            do_valid_callback(x + x_delta * tile, y - y_offset + y_delta * tile, offset_north + tile, callback);
-            do_valid_callback(x - x_delta * tile, y - y_offset + y_delta * tile, offset_north + tile * GRID_SIZE, callback);
-            do_valid_callback(x + x_delta * tile, y + y_offset - y_delta * tile, offset_south - tile * GRID_SIZE, callback);
-            do_valid_callback(x - x_delta * tile, y + y_offset - y_delta * tile, offset_south - tile, callback);
+            do_valid_callback(x + x_delta * tile, y - y_offset + y_delta * tile, move_grid_offset_by_delta(grid_offset, tile + offset_north, offset_north), callback);
+            do_valid_callback(x - x_delta * tile, y - y_offset + y_delta * tile, move_grid_offset_by_delta(grid_offset, offset_north, tile + offset_north), callback);
+            do_valid_callback(x + x_delta * tile, y + y_offset - y_delta * tile, move_grid_offset_by_delta(grid_offset, offset_south, offset_south - tile), callback);
+            do_valid_callback(x - x_delta * tile, y + y_offset - y_delta * tile, move_grid_offset_by_delta(grid_offset, offset_south - tile, offset_south), callback);
```

Comment:
Please break up the lines so they're not too long: breaking just before the `move_...` function calls should be good.